src/main/scala/mesosphere/marathon/api/akkahttp/v2/PodsController.scala

```diff
@@ -180,4 +209,31 @@ class PodsController(
     }
   // format: ON
 
+  private def onSuccessLegacy[T](maybePodId: Option[PathId])(f: => Future[T])(implicit identity: Identity): Directive1[T] = onComplete({
```

Comment:
Don't we have this for apps already?

Reply:
+1 - if it is the same (and I guess it is) I wouldn't copy-paste it...

Reply:
Yep, but I'd prefer not to share the code here.